File: backend/core/src/modules/points/points.controller.ts
import { Controller, Get, Req, UseGuards } from '@nestjs/common';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { PointsService } from './points.service';

@UseGuards(JwtAuthGuard)
@Controller('points')
export class PointsController {
  constructor(private readonly pointsService: PointsService) {}

  @Get('balance')
  balance(@Req() req: any) {
    return this.pointsService.balance(req.user?.userId);
  }

  @Get('transactions')
  transactions(@Req() req: any) {
    return this.pointsService.transactions(req.user?.userId);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/PointsController.java
package store.pjcloud.cowrite.core.controller;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.entity.PointTransaction;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.PointsService;

@RestController
@RequestMapping("/points")
public class PointsController {
    private final PointsService pointsService;

    public PointsController(PointsService pointsService) {
        this.pointsService = pointsService;
    }

    @GetMapping("/balance")
    public Map<String, Object> balance() {
        UUID userId = SecurityUtils.requireUserId();
        return pointsService.balance(userId);
    }

    @GetMapping("/transactions")
    public List<PointTransaction> transactions() {
        UUID userId = SecurityUtils.requireUserId();
        return pointsService.transactions(userId);
    }
}
